samples/algorithms/sudoku-grover/SudokuQuantum.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+
+namespace Microsoft.Quantum.Samples.SudokuGrover
+{
+    /// # Info
+    /// Code to solve a Sudoku puzzle by transforming it into a graph problem 
+    /// and calling the Quantum SolvePuzzle operation to solve it
+    class SudokuQuantum 
+    {
+        /// # Summary
+        /// QuantumSolve will call Q# code to solve the Sudoku puzzle and display the solution.
+        /// # Input
+        /// ## puzzle
+        /// The Sudoku puzzle to solve.
+        /// ## sim
+        /// The quantum simulator.
+        public static bool QuantumSolve(int[,] puzzle, QuantumSimulator sim)
```

Comment:
```suggestion
        public static bool QuantumSolve(int[,] puzzle, SimulatorBase sim)
```

Reply:
Another suggestion from @cgranade is to make this function async: https://github.com/microsoft/Quantum/pull/403#discussion_r473471765